programs/stake/src/stake_state.rs

```diff
@@ -1470,7 +1480,7 @@ impl MergeKind {
             (Self::Inactive(_, _), Self::ActivationEpoch(_, _)) => None,
             (Self::ActivationEpoch(meta, mut stake), Self::Inactive(_, source_lamports)) => {
                 stake.delegation.stake = checked_add(stake.delegation.stake, source_lamports)?;
-                Some(StakeState::Stake(meta, stake))
+                Some(StakeState::Stake(meta, stake, DeactivationFlag::Empty))
```

Comment:
I think this still allows the problem unfortunately. If I redelegate from stake A to B, and then merge C into B, this would clear the deactivation flag.

I appreciate that it's more work, but could you add some test cases for merging a redelegated stake? Then we'll be sure we're ok

Reply:
Nice catch. Should be fixed now.